### ISSUE TITLE
Segfaults after local install on Mac

### ISSUE THREAD

**Issue by jcjohnson** [facebookresearch/Aria_data_tools#4]
I have installed Aria Data Tools locally on my Macbook (Intel processor, macOS 12.3.1) in order to visualize sequences from the Pilot Dataset.

I couldn't find much documentation on how to install on Mac, so I manually stepped through the [Dockerfile](https://github.com/facebookresearch/Aria_data_tools/blob/main/Dockerfile) and duplicated steps one by one.

I got everything to build and the tests appear to pass (Running `ctest -j` from my cmake build directory reports 100% tests pass). I can also import the datatools in a Python interactive session (`import pyark.datatools as datatools`).

I'm now trying to run the [Visualization Tool](https://facebookresearch.github.io/Aria_data_tools/docs/howto/examples/#visualizing-sequences-and-pre-computed-camera-trajectory) for sequences from the pilot dataset, like this:

```bash
cd src/visualization
python main.py --vrs_path [path to VRS file]
```

On some sequences from the Pilot Dataset (such as `sample_dataset/desktop_activity/aria_recording.vrs`) the GUI appears, but the program crashes with a segfault and no error message when I click the play button. On other recordings (such as `sample_dataset/everyday_activity/recording_2/recording.vrs`) the GUI appears and I can play the file, but the program crashes with a segfault (and no other error message) at a seemingly random point during playback.

Is this expected behavior? Is there any way to tell what is going wrong?

**Comment by zhaoyang-lv** [comment]
First to verify, is the docker environment working well for you on Mac? 
We previously manually deprioritize the manual building instructions on Mac because we have not made the building process working for different Mac in our internal test. Some frictions are expected. But I don't think we have a seen a similar segfaul issue at runtime with Mac.

**Comment by jcjohnson** [comment]
I haven't tried the docker install, but I'll give it a try and report back.

**Comment by rawalkhirodkar** [comment]
If you want to try, I have a [docker](https://hub.docker.com/r/rawalkhirodkar/aria_data_tools) container which works for Mac.

**Comment by zhaoyang-lv** [comment]
Please try the visualizer in the docker on the everyday activity first "sample_dataset/everyday_activity/recording_2/recording.vrs".
The visualizer in theory should also work with desktop activity in the docker image. We have a pending change to visualizer specifically for that activity so that we load all the files together.

**Comment by fskarakas** [comment]
Project Aria recordings in `desktop_activities` folder do not contain SLAM camera streams, but `AriaViewer` class used by `main.py` expect them to fetch information and display images. This explains the first issue discussed above. A viewer for playing desktop activities recordings is soon to be released as @zhaoyang-lv mentioned.

Found out that the second issue - segmentation fault with a recording from `everyday_activity` folder -  occurs when playback is paused, then resumed again. It is related to how the current timestamp gets modified in the loop that reads the records. Thanks for the heads up, testing a fix now.

**Comment by davidsonic** [comment]
I pull @rawalkhirodkar's docker image and followed the documentation https://facebookresearch.github.io/Aria_data_tools/docs/howto/calibration/. The environment seems to work well with the first few commands but the CalibrationModel cannot be loaded:
```
p_slamLeft = np.array([3.0, 2.0, 1.0])
uv_slamLeft = device_model.getCameraCalib('camera-slam-left').projectionModel.project(p_slamLeft)
```
which gives the error:
```
device model <pyark.datatools.sensors.DeviceModel object at 0x4001277570>
Traceback (most recent call last):
  File "project_unproject.py", line 24, in <module>
    uv_slamLeft = device_model.getCameraCalib('camera-slam-left').projectionModel.project(p_slamLeft)
AttributeError: 'NoneType' object has no attribute 'projectionModel'
```

**Comment by davidsonic** [comment]
Following the above post, I found that the project/unproject functions yield strange behaviors with different inputs. 
I tested three inputs, and it looks like that unproject does not always invert project?
```
p_rgb [3. 2. 1.]
uv_rgb [2134.58140337 1650.95653425]
unproject_rgb [3. 2. 1.]
===========================
p_rgb [4. 8. 9.]
uv_rgb [ 957.93369855 1198.65883829]
unproject_rgb [0.44444444 0.88888889 1.        ]
===========================
p_rgb [ 8 13 15]
uv_rgb [1005.43497452 1183.58550113]
unproject_rgb [0.53333333 0.86666667 1.        ]

```
The code to reproduce is below:
```
import pyark.datatools as datatools
import numpy as np

filename = '/data/recording_1/recording.vrs'
vrs_data_provider = datatools.dataprovider.AriaVrsDataProvider()
load = vrs_data_provider.openFile(filename)
if load: 
    print('loading successful')
else:
    print('failed to load')
    exit()

camera_rgb_type_id = 214
camera_rgb_instance_id = 1
camera_rgb_stream_id = datatools.dataprovider.StreamId(camera_rgb_type_id, camera_rgb_instance_id)
vrs_data_provider.setStreamPlayer(camera_rgb_stream_id)
vrs_data_provider.readFirstConfigurationRecord(camera_rgb_stream_id)
vrs_data_provider.loadDeviceModel()
device_model = vrs_data_provider.getDeviceModel()

p_rgb = np.array([3.0, 2.0, 1.0])
print('p_rgb', p_rgb)
uv_rgb = device_model.getCameraCalib('camera-rgb').projectionModel.project(p_rgb)
print('uv_rgb', uv_rgb)
rep_rgb = device_model.getCameraCalib('camera-rgb').projectionModel.unproject(uv_rgb)
print('unproject_rgb', rep_rgb)
```

**Comment by gustavomart** [comment]
Issue was fixed by [56f185a](https://github.com/facebookresearch/Aria_data_tools/commit/56f185ad392482c26ce22a379db91ee5b62a0cd7)
